cmd/fry.go

```diff
@@ -19,69 +19,132 @@ import (
 	"errors"
 	"fmt"
 	"os"
+	"strings"
 
+	"github.com/sirupsen/logrus"
 	"github.com/sonatype-nexus-community/hashbrowns/iq"
+	"github.com/sonatype-nexus-community/hashbrowns/logger"
 	"github.com/sonatype-nexus-community/hashbrowns/parse"
 	"github.com/sonatype-nexus-community/hashbrowns/types"
 
 	"github.com/sonatype-nexus-community/nancy/cyclonedx"
 	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
 )
 
+var log *logrus.Logger
+
 // fryCmd represents the fry command
 var fryCmd = &cobra.Command{
 	Use:   "fry",
 	Short: "Submit list of sha1s to Nexus IQ Server",
 	Long: `Provided a path to a file with sha1's and locations, this command will submit them to Nexus IQ Server.
 
 This can be used to audit generic environments for matches to known hashes that do not meet your org's policy.`,
+	SilenceErrors: true,
 	RunE: func(cmd *cobra.Command, args []string) (err error) {
+		defer func() {
+			if r := recover(); r != nil {
+				var ok bool
+				err, ok = r.(error)
+				if !ok {
+					err = fmt.Errorf("pkg: %v", r)
+				}
+
+				logger.PrintErrorAndLogLocation(err)
```

Comment:
+1, Very nice. Do we need it at the root cmd? (Don't know, just wondering) Probably not, since you're not really doing much until you try to execute the `fry` command.

I'm on the fence about the whole `panic() then recover()` verses `return error` approach, but let's give it a whirl.

Reply:
I think this is a good use of the `recover` functionality. I think the most basic use of `recover` is to ultimately prevent your application from coming crashing down completely, and that's how I'm using it (and to avoid spurious uses of `os.Exit`).